Numpy bool msgpack bugfix

PhyNerd opened the pull request:
This fixes issue #18390.
It allows for running DataFrame.to_msgpack() with dataframes containing fields of the numpy.bool_ datatype. 

- [x] closes #18390
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

PhyNerd committed:
Fixed numpy bool serialization to msgpack

This allows for running to_msgpack with dataframes containing field of numpy.bool_ datatype

PhyNerd committed:
Added to test

PhyNerd committed:
Added whatsnew entry

jreback commented on pandas/io/msgpack/_packer.pyx:
```diff
@@ -133,7 +134,7 @@ cdef class Packer(object):
             if o is None:
                 ret = msgpack_pack_nil(&self.pk)
-            elif isinstance(o, bool):
+            elif isinstance(o, (bool, bool_)):
```
use ``np.bool_``

PhyNerd replied:
Will do.

jreback commented on pandas/tests/io/msgpack/test_pack.py:
```diff
@@ -25,6 +26,7 @@ def testPack(self):
             (), ((),), ((), None,),
             {None: 0},
             (1 << 23),
```
can you add a round-trip test for the example in the issue itself.

PhyNerd replied:
This test should already test for the buggy behavior when it comes to `np.bool_`.

Should I revert the changes to this test and add a one checking round-trip behavior or should I make that test a additional one? 
Where would I best add this test? (There is no /test/io/formats/test_to_msgpack.py)

PhyNerd committed:
Added round-trip Tests, changend bool_ to np.bool_

Added test for scaler numpy bool, added numpy bool to mixed list test, removed modifications to pack test and changend bool_ to np.bool_

PhyNerd commented:
I applied the requested changes. 
I'm not sure why but one of the Travis CI builds timed-out.

jreback commented:
can you add the tests from the original issue as well, e.g. the round-trip of the object array containing bools. rebase and ping on green.

PhyNerd committed:
Added test for mixed Series with numpy bool

PhyNerd commented:
I added the requested check.

jreback commented:
lgtm. just needs a rebase. ping when pushed.

PhyNerd commented:
rebased.

jreback commented:
thanks @PhyNerd
